quantum/split_common/transactions.c

```diff
@@ -53,6 +67,36 @@ void slave_rpc_exec_callback(uint8_t initiator2target_buffer_size, const void *i
 ////////////////////////////////////////////////////
 // Helpers
 
+bool transport_transaction(int8_t id, const void *initiator2target_buf, uint16_t initiator2target_length, void *target2initiator_buf, uint16_t target2initiator_length) {
+#if SPLIT_MAX_CONNECTION_ERRORS < 0
+    return transport_execute_transaction(id, initiator2target_buf, initiator2target_length, target2initiator_buf, target2initiator_length);
+#else   // SPLIT_MAX_CONNECTION_ERRORS < 0
+    // Throttle transaction attempts if target doesn't seem to be connected
+    // Without this, a solo half becomes unusable due to constant read timeouts
+    static uint8_t  connection_errors      = 0;
+    static uint16_t connection_check_timer = 0;
+    if (connection_errors >= SPLIT_MAX_CONNECTION_ERRORS && timer_elapsed(connection_check_timer) < SPLIT_CONNECTION_CHECK_TIMEOUT) {
+        return false;
+    }
+    connection_check_timer = timer_read();
```

Comment:
(I only mean line 81 the timer_read())
You could move this into the error disconnect condition in line 88. As I read the code, it indicates the time of last successful connection and as such the timer_read() doesn't need to be in the happy path, saving some cycles. Moving it there would start counting from the time it really failed, which is okay?

Reply:
Hmm. True.